src/languages/singlestoredb/singlestoredb.formatter.ts

```diff
@@ -0,0 +1,267 @@
+  'CREATE ROLE',
+  'CREATE USER',
+  'CREATE VIEW',
+  'DEALLOCATE PREPARE',
+  'DESCRIBE',
+  'DETACH DATABASE',
+  'DETACH PIPELINE',
+  'DO',
+  'DROP DATABASE',
+  'DROP FUNCTION',
+  'DROP INDEX',
+  'DROP LINK',
+  'DROP PIPELINE',
+  'DROP PROCEDURE',
+  'DROP RESOURCE POOL',
+  'DROP ROLE',
+  'DROP USER',
+  'DROP VIEW',
+  'EXECUTE',
+  'EXPLAIN',
+  'FLUSH',
+  'FORCE',
+  'GRANT',
+  'HANDLER',
+  'HELP',
+  'KILL CONNECTION',
+  'KILLALL QUERIES',
+  'LOAD DATA',
+  'LOAD INDEX INTO CACHE',
+  'LOAD XML',
+  'LOCK INSTANCE FOR BACKUP',
+  'LOCK TABLES',
+  'MASTER_POS_WAIT',
+  'OPTIMIZE TABLE',
+  'PREPARE',
+  'PURGE BINARY LOGS',
+  'REBALANCE PARTITIONS',
+  'RELEASE SAVEPOINT',
+  'REMOVE AGGREGATOR',
+  'REMOVE LEAF',
+  'REPAIR TABLE',
+  'REPLACE',
+  'REPLICATE DATABASE',
+  'RESET',
+  'RESET MASTER',
+  'RESET PERSIST',
+  'RESET REPLICA',
+  'RESET SLAVE',
+  'RESTART',
+  'RESTORE DATABASE',
+  'RESTORE REDUNDANCY',
+  'REVOKE',
+  'ROLLBACK',
+  'ROLLBACK TO SAVEPOINT',
+  'SAVEPOINT',
+  'SET CHARACTER SET',
+  'SET DEFAULT ROLE',
+  'SET NAMES',
+  'SET PASSWORD',
+  'SET RESOURCE GROUP',
+  'SET ROLE',
+  'SET TRANSACTION',
+  'SHOW',
+  'SHOW CHARACTER SET',
+  'SHOW COLLATION',
+  'SHOW COLUMNS',
+  'SHOW CREATE DATABASE',
+  'SHOW CREATE FUNCTION',
+  'SHOW CREATE PIPELINE',
+  'SHOW CREATE PROCEDURE',
+  'SHOW CREATE TABLE',
+  'SHOW CREATE USER',
+  'SHOW CREATE VIEW',
+  'SHOW DATABASES',
+  'SHOW ENGINE',
+  'SHOW ENGINES',
+  'SHOW ERRORS',
+  'SHOW FUNCTION CODE',
+  'SHOW FUNCTION STATUS',
+  'SHOW GRANTS',
+  'SHOW INDEX',
+  'SHOW MASTER STATUS',
+  'SHOW OPEN TABLES',
+  'SHOW PLUGINS',
+  'SHOW PRIVILEGES',
+  'SHOW PROCEDURE CODE',
+  'SHOW PROCEDURE STATUS',
+  'SHOW PROCESSLIST',
+  'SHOW PROFILE',
+  'SHOW PROFILES',
+  'SHOW RELAYLOG EVENTS',
+  'SHOW REPLICA STATUS',
+  'SHOW REPLICAS',
+  'SHOW SLAVE',
+  'SHOW SLAVE HOSTS',
+  'SHOW STATUS',
+  'SHOW TABLE STATUS',
+  'SHOW TABLES',
+  'SHOW VARIABLES',
+  'SHOW WARNINGS',
+  'SHUTDOWN',
+  'SNAPSHOT DATABASE',
+  'SOURCE_POS_WAIT',
+  'START GROUP_REPLICATION',
+  'START PIPELINE',
+  'START REPLICA',
+  'START SLAVE',
+  'START TRANSACTION',
+  'STOP GROUP_REPLICATION',
+  'STOP PIPELINE',
+  'STOP REPLICA',
+  'STOP REPLICATING',
+  'STOP SLAVE',
+  'TABLE',
+  'TEST PIPELINE',
+  'TRUNCATE TABLE',
+  'UNLOCK INSTANCE',
+  'UNLOCK TABLES',
+  'USE',
+  'WITH',
+  'XA',
+  // flow control
+  // 'IF',
+  'ITERATE',
+  'LEAVE',
+  'LOOP',
+  'REPEAT',
+  'RETURN',
+  'WHILE',
+  // other
+  'ADD',
+  'WINDOW',
+  'PARTITION BY',
+]);
+
+const reservedSetOperations = expandPhrases(['UNION [ALL | DISTINCT]']);
+
+const reservedJoins = expandPhrases([
+  'JOIN',
+  '{LEFT | RIGHT | FULL} [OUTER] JOIN',
+  '{INNER | CROSS} JOIN',
+  'NATURAL {LEFT | RIGHT} [OUTER] JOIN',
+  // non-standard joins
+  'STRAIGHT_JOIN',
+]);
+
+// https://dev.mysql.com/doc/refman/8.0/en/
+export default class SingleStoreDBFormatter extends Formatter {
```

Comment:
Should rename to:
```suggestion
export default class SingleStoreDbFormatter extends Formatter {
```

For consistency with other formatter classes like `Db2Formatter`, `MariaDbFormatter`.